### PR TITLE
Caching JSON-LD from bibdata for merged JSON-LD view

### LINKED ISSUE #1059: Expose bibrecord JSON-LD combined with local plum record JSON-LD.
Do we want to cache a fork of the JSON-LD or go to bibdata every time it's requested?

### PR DESCRIPTION
**Pull request by escowles**
Fixes #1059

### CODE REVIEW

File: app/models/solr_document.rb [hunk 1]
```diff
@@ -97,6 +109,10 @@ def collection
     self[Solrizer.solr_name('member_of_collections', :symbol)]
   end
 
+  def collection_ids
```
Comment (tpendragon) [inline app/models/solr_document.rb]: I think we actually want the collection names indexed.

Comment (escowles) [inline app/models/solr_document.rb]: OK, I'll update this.

**Comment by tpendragon** [comment]
Does this provide a view too?

**Comment by escowles** [comment]
There is an existing view in Hyrax, you can append `.jsonld` to the show URL (though that's broken in Plum currently).  The override of `export_as_jsonld` is what drives that view, see https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/curation_concern_controller.rb#L71-L79 — we may want to do TTL and NT too.

**Comment by tpendragon** [comment]
Cool. Can you add a new ticket to fix the jsonld view, since this is gonna close #1059?

**Comment by escowles** [comment]
Oh, this PR fixes the JSON-LD view — the default implementation of `export_as_jsonld` is broken, but this overridden version fixes that.

**Comment by tpendragon** [comment]
Oh, okay. Does this build up a valid JSON-LD document for, IE, the memberOf predicate? Since Pom loads the graph.

**Comment by tpendragon** [comment]
Oh, it actually doesn't load the graph! So it'll work even if it's not valid JSON-LD.

**Comment by tpendragon** [comment]
Can we get a ticket for appending our local terms to the context? Then I'll approve and merge after CI passes.

**Comment by escowles** [comment]
Right now, I think it's valid JSON-LD, but not correct RDF:

`"memberOf": "Test Collection"`

instead of something like:

`"memberOf":{"title": "Test Collection"}`

**Comment by escowles** [comment]
Re: updating the context, I've opened a PR for that (and adding sort_title and the ARK to the bibdata output): https://github.com/pulibrary/marc_liberation/pull/217

**Comment by escowles** [comment]
This PR now supports JSON-LD, Turtle and NTriples, and generates objects for each Collection instead of incorrectly using `pcdm:memberOf "title"`.

There are also some updates to work around changes in the source data compared to what was in the VCR cassette.  I decided to create a new cassette for the linked data tests to get the updated bibdata context.json file, and that brought along some data changes to the sample record too.

File: app/presenters/hyrax_show_presenter.rb [hunk 2]
```diff
@@ -16,8 +16,44 @@ def member_presenter_factory
...
+        identifier: identifier
+      }.reject { |_, v| v.nil? || v.try(:empty?) }
+    end
+
+    def url
+      Rails.application.routes.url_helpers.send "hyrax_#{model_name.singular}_url", id
+    end
+
+    def collection_objects
+      collection.map { |title| { title: title } }
```
Comment (tpendragon) [inline app/presenters/hyrax_show_presenter.rb]: I hate to bring this up, but I wonder if this should include the @id for the collection too?

**Comment by escowles** [comment]
An example:

```sh
$ curl -H "Accept: text/turtle" http://localhost:3000/concern/multi_volume_works/dc247ds08n

<http://localhost:3000/concern/multi_volume_works/dc247ds08n> <http://purl.org/dc/terms/title> "秋塍三州詩鈔 : 四卷";
   <http://iiif.io/api/presentation/2#navDate> "2017-03-29T00:00:00Z";
   <http://pcdm.org/models#memberOf> [ <http://purl.org/dc/terms/title> "East Asian Library Digital Bookshelf"];
   <http://purl.org/dc/elements/1.1/identifier> "ark:/88435/2j62s7595";
   <http://www.europeana.eu/schemas/edm/rights> "http://rightsstatements.org/vocab/NKC/1.0/";
   <http://www.w3.org/2004/02/skos/core#scopeNote> "just the good parts" .


$ curl -H "Accept: application/ld+json" http://localhost:3000/concern/multi_volume_works/dc247ds08n

{"title":"秋塍三州詩鈔 : 四卷",
 "@context":"https://bibdata.princeton.edu/context.json",
 "@id":"http://localhost:3000/concern/multi_volume_works/dc247ds08n",
 "memberOf":[{"title":"East Asian Library Digital Bookshelf"}],
 "scopeNote":["just the good parts"],
 "navDate":["2017-03-29T00:00:00Z"],
 "edm_rights":["http://rightsstatements.org/vocab/NKC/1.0/"],
 "identifier":"ark:/88435/2j62s7595"}
```

**Comment by escowles** [comment]
OK, the collection objects are identified now:

```
{
  "title": "秋塍三州詩鈔 : 四卷",
  "@context": "https://bibdata.princeton.edu/context.json",
  "@id": "http://localhost:3000/concern/multi_volume_works/dc247ds08n",
  "memberOf": [
    {
      "@id": "http://localhost:3000/collections/dcr56n099c",
      "title": "East Asian Library Digital Bookshelf"
    }
  ],
  "scopeNote": [
    "just the good parts"
  ],
  "navDate": [
    "2017-03-29T00:00:00Z"
  ],
  "edm_rights": [
    "http://rightsstatements.org/vocab/NKC/1.0/"
  ],
  "identifier": "ark:/88435/2j62s7595"
}
```

**Pull request merged**